src/components/live-kit-room/live-kit-room.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  Component,
+  Host,
+  h,
+  Prop,
+  Element,
+  Watch,
+  forceUpdate
+} from "@stencil/core";
+import { LiveKitCallbacks } from "./types";
+import { connectToRoom } from "./connect";
+import { Participant, RemoteParticipant, Room, Track } from "livekit-client";
+import { removeElement } from "../../common/array";
+
+/**
+ * TODO.
+ */
+@Component({
+  tag: "ch-live-kit-room",
+  styleUrl: "live-kit-room.scss",
+  shadow: true
+})
+export class ChLiveKitRoom {
+  @Element() el!: HTMLChLiveKitRoomElement;
+
+  #participants: {
+    participant: Participant | RemoteParticipant;
+    ref?: HTMLAudioElement;
+    shouldUpdate?: boolean;
+  }[] = [];
+
+  #currentRoom: Room;
+
+  /**
+   * Specifies the callbacks required in the control.
+   */
+  @Prop() readonly callbacks?: LiveKitCallbacks | undefined;
+
+  /**
+   * TODO
+   */
+  @Prop() readonly url: string = "";
+
+  /**
+   * TODO
+   */
+  @Prop() readonly token: string = "";
+
+  /**
+   * Specifies the room state.
+   */
+  @Prop() readonly connected: boolean = false;
+
+  @Watch("connected")
+  connectedChanged() {
+    if (this.connected) {
+      this.#connect();
+    } else {
+      this.#disconnectRoom();
+    }
+  }
+
+  /**
+   * Specifies the room state.
+   */
+  @Prop() readonly micEnable: boolean = false;
```

Comment:
Don't shorten property names. They have to be descriptive.

```suggestion
  @Prop() readonly microphoneEnabled: boolean = false;
```